### DISCUSSION TITLE
Wrote a CNG-compatible Expo config plugin to install this, should we add it to the docs?

### DISCUSSION THREAD

**Discussion by davidalejandroaguilar** [react-native-maps/react-native-maps#5301] [Feature requests]
I wrote an Expo config plugin https://www.npmjs.com/package/react-native-maps-expo-plugin to do the iOS installation instructions for Google Maps without users having to manually tweak files, since doing so would effectively prevent users from using CNG (here's Expo documentation on why tweaking the files yourself is not a good idea): [https://docs.expo.dev/workflow/overview/#what-if-i-want-to-edit-the](https://docs.expo.dev/workflow/overview/#what-if-i-want-to-edit-the).

The plugin will basically do the `react-native-maps` installation instructions for users every time, so that they can keep enjoying CNG.

I'm a new React Native user, so I wrote this after sleeping until 2 AM figuring out why I couldn't get `react-native-maps` to run on a new app. After figuring out why (will open a separate PR to improve installation instructions) and reading about Expo's CNG, I noticed that the installation instructions effectively prevented users from using CNG (some users even [reported losing their config](https://github.com/react-native-maps/react-native-maps/issues/5049#issuecomment-2277731001)).

I don't want to open a PR kind of self-promoting or anything, so first I'm opening a discussion to see if y'all think this will benefit the community.

**Comment by salah-ghanim** [comment]
@davidalejandroaguilar thank you for creating this plugin, I have to admit I have no experience in expo / expo go so I can't judge how much of this should be part of the repo or how much of it should stay as it's own plugin / repo.

I'm fully focused at the moment on moving to the new architecture and have no capacity to look into this.

but once done I will look into expo integration, it would make sense to try and make it as smooth as possible.

**Comment by StefanWallin** [comment]
Seems like it already should work with `npx expo install`?

https://docs.expo.dev/versions/latest/sdk/map-view/

Is this discussion out of date now?

**Comment by salah-ghanim** [comment]
I believe it's out of date, we just merged an expo plugin in a recent release [v1.22.0](https://github.com/react-native-maps/react-native-maps/releases/tag/v1.22.0)
